### PR TITLE
use lcov+cmake instead of gcovr

### LINKED ISSUE #80: Use cmake+lcov instead of gcovr for coverage
Our coverage reports are off because of some weird partial hits reported by gcovr. I don't know if it is possible to fully disable branch detection with gcovr, but with lcov this option is enabled by default so hopefully it generates more useful coverage reports. Also, cpp/cmake/CodeCoverage.cmake generates a cmake target for creating coverage reports with lcov, so it's worth to give this a try.

### PR DESCRIPTION
**Pull request by joergbrech**
## Merge Request - GuideLine Checklist 

Fixes #80 

**Guideline** to check code before resolve WIP and approval, respectively.
As many checkboxes as possible should be ticked.

### Checks by code author:
* [x] There is at least one issue associated with the pull request.
* [ ] The branch follows the naming conventions as defined in the [git workflow](git-workflow).
* [ ] New code adheres with the [coding guidelines](coding-guidelines)
* [ ] Tests for new functionality has been added
* [ ] A local test was succesful
* [ ] There is appropriate **documentation** of your work. (use doxygen style comments)
* [ ] If new third party software is used, did you pay attention to its license? Please remember to add it to the wiki after successful merging.
* [ ] If new mathematical methods or epidemiological terms are used, has the glossary been updated ? Did you provide further documentation ?
 is present or referenced. Please provide your references.
* [ ] The following questions are addressed in the documentation*:  Developers (what did you do?, how can it be maintained?), For users (how to use your work?), For admins (how to install and configure your work?)
* For documentation: Please write or update the Readme in the current working directory!

### Checks by code reviewer(s):
* [ ] Is the code clean of development artifacts e.g., unnecessary comments, prints, ...
* [ ] The ticket goals for each associated issue are reached or problems are clearly addressed (i.e., a new issue was introduced).
* [ ] There are appropriate **unit tests** and they pass.
* [ ] The git history is clean and linearized for the merge request.
* [ ] Coverage report for new code is acceptable.

### CODE REVIEW

**Comment by joergbrech** [comment]
Code Coverage: 90,74% 😎

**Review by dabele — CHANGES_REQUESTED** [review]
Great. Noone has managed +27% coverage before :smile:. I was wondering why there is this lcov target in cmake that isn't used anywhere.

It's a bit of a shame that there seems to be no good way to check coverage for lines like `if (a || b)` without getting all the exception branches as well. For example, if it is something like `p != nullptr || *p == 0)` and it is only tested with non-null, there could be issues if the code afterwards uses p without checking. It would be possible to compile most of our code (probably everything that isn't IO) without exceptions, but not sure this is worth the effort.

File: .github/actions/linux-test/action.yml [hunk 1]
```diff
@@ -32,29 +30,27 @@ runs:
...
+          sudo chmod a+x runUnitTests
+          ./runUnitTests  --gtest_output="xml:report.xml"
         fi
     - name: Upload test report
       uses: actions/upload-artifact@v2
       with:
         name: test-cpp-linux-report
         path: cpp/build/tests/report.xml
-        if-no-files-found: error
+        if-no-files-found: ignore
```
[the remark this answers is not in the document]
Comment (joergbrech) [reply .github/actions/linux-test/action.yml:46]: > just to be sure, the tests are still run and appear in the job log and the job still fails if the tests fail, but the artifact is not created?

I haven't tested all of the branches in that sentence (hehe), especially the failed tests ones, so I am not 100% sure. But I assumed that that is how it works.

I will commit your suggestion, lets see if it works. If yes, I will change the other artifact as well.

**Comment by joergbrech** [comment]
>  It would be possible to compile most of our code (probably everything that isn't IO) without exceptions, but not sure this is worth the effort.

 Yeah, code coverage for IO is probably more valueable than branch-detection without exceptions...

File: .github/actions/linux-test/action.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (joergbrech) [reply .github/actions/linux-test/action.yml]: Works!

**Review by dabele — APPROVED** [review]

**Pull request merged**